resources/sugarcrm/sugarprotocolbase.h

```diff
@@ -21,13 +21,17 @@
 #ifndef SUGARPROTOCOLBASE_H
 #define SUGARPROTOCOLBASE_H
 
-#include<QString>
+#include <sugarsession.h>
+#include <QString>
 
 class SugarProtocolBase
 {
 public :
     virtual ~SugarProtocolBase();
     virtual int login(const QString &user, const QString &password, QString &sessionId, QString &errorMessage) = 0;
+    virtual void setSession(SugarSession *session) = 0;
+protected :
+    SugarSession *mSession;
```

Comment:
Since you're not using it in the mock (and in order to keep this base class as pure interface), you should move it to the Soap subclass.